sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/TopDocumentQueryExecutionContext.java

```diff
@@ -138,4 +124,45 @@ public boolean test(FeedResponse<T> frp) {
             }
         });
     }
+
+    // TODO: This could be restructured eventually with better design to avoid these many checks
```

Comment:
hmm, maybe can use recursion? so I think the following can be changed to:

       private ParallelDocumentQueryExecutionContextBase<T> getContext(IDocumentQueryExecutionComponent<T> component) {
        if (component instanceof GroupByDocumentQueryExecutionContext) {
            return this.getContext(((GroupByDocumentQueryExecutionContext<T>) component).getComponent());
        } if (component instanceof DistinctDocumentQueryExecutionContext) {
            return this.getContext(((DistinctDocumentQueryExecutionContext<T>) component).getComponent());
        } else if (component instanceof AggregateDocumentQueryExecutionContext) {
            return this.getContext(((AggregateDocumentQueryExecutionContext<T>) component).getComponent());
        } else if (component instanceof SkipDocumentQueryExecutionContext) {
            return this.getContext(((SkipDocumentQueryExecutionContext<T>) component).getComponent());
        } else {
            return (ParallelDocumentQueryExecutionContextBase<T>) component;
        }
    }

Reply:
Wanted to avoid recursion deliberately to avoid the extra cost involved in recursion vs few lines of code. Also, we need to redesign the to avoid these checks all together.

Reply:
I see, thanks for the explanation~